syntax = "proto3";

package audio;

service AudioIngestor {
  rpc SendAudio (AudioChunk) returns (Ack);
}

message AudioChunk {
  string device_id = 1;
  string seq_id = 2;
  int64 timestamp_ms = 3;
  int32 sample_rate = 4;
  bytes wav_data = 5;
}

message Ack {
  bool success = 1;
  string seq_id = 2;
}
